lib/observer.js

```diff
@@ -88,11 +88,59 @@ ObserverMixin.notifyObserversOf = function(operation, context, callback) {
     );
   });
   return callback.promise;
-}
+};
 
 ObserverMixin._notifyBaseObservers = function(operation, context, callback) {
   if (this.base && this.base.notifyObserversOf)
     this.base.notifyObserversOf(operation, context, callback);
   else
     callback();
-}
+};
+
+/**
+ * Run the given function with before/after observers. It's done in three serial
+ * steps asynchonously:
+ * 
+ * - Notify the registered observers under 'before ' + operation
+ * - Execute the function
+ * - Notify the registered observers under 'after ' + operation
+ *
+ * If an error happens, it fails fast and calls the callback with err.
+ *
+ * @param {String} operation The operation name
+ * @param {Context} context The context object
+ * @param {Function} fn The task to be invoked as fn(done) or fn(context, done)
+ * @param {Function} callback The callback function
+ * @returns {*}
+ */
+ObserverMixin.around = function(operation, context, fn, callback) {
```

Comment:
shouldn't this method name conform to the `notifyObserversOf` convention?

Consider this idea (in the context of your usage):

```js
function work(done) {
  self.connector.commit(self.connection, done);
}

self.notifyObserversBeforeAndAfter('commit', context, work, function(err) {
  // Deference the connection to mark the transaction is not active
  // The connection should have been released back the pool
  self.connection = null;
  cb(err);
});
```